src/opsinputs/opsinputs_fill_mod.F90

```diff
@@ -1434,24 +1436,122 @@ subroutine opsinputs_fill_fillreal2dfromgeoval( &
 type(opsinputs_jeditoopslayoutmapping), intent(in) :: JediToOpsLayoutMapping
 real(c_double), intent(in)                         :: hofx(:, :)
 type(oops_variables), intent(in)                   :: varnames
+character(len=*), intent(in), optional             :: JediObsName
+logical, intent(in), optional                      :: WriteHofXIntoCX
 
 ! Local declarations:
 integer :: i
 integer :: hofxIndex
+logical :: hofxToCx
+
+hofxToCx = .false.
+if (present(WriteHofXIntoCX)) then
+   hofxToCx = WriteHofXIntoCX
+end if
 
 ! Body:
 
 if (JediToOpsLayoutMapping % ConvertRecordsToMultiLevelObs) then
-   call opsinputs_fill_fillreal2dfromgeovalformultilevelobs( &
-        Hdr, OpsVarName, JediToOpsLayoutMapping % NumOpsObs, Real2, &
-        GeoVals, GeoVaLsAreTopToBottom, JediVarName, JediToOpsLayoutMapping)
+   ! Index of variable in hofx array.
+   hofxIndex = 0
+   do i = 1, size(hofx, 1)
+      if (present(JediObsName)) then
```

Comment:
The use of `JediObsName` reproduces the name mapping between GeoVaL and observation names (e.g. relative_humidity -> relativeHumidity). This wasn't needed in the code removed by #110.